src/Schema.php

```diff
@@ -1458,6 +1462,62 @@ protected function validateDatetime($value, ValidationField $field) {
         return $value;
     }
 
+    /**
+     * Recursively resolve allOf inheritance tree and return a merged resource specification
+     *
+     * @param ValidationField $field The validation results to add.
+     * @return array Returns an array of merged specs.
+     * @throws ParseException Throws an exception if an invalid allof member is provided
+     * @throws RefNotFoundException Throws an exception if the array has an items `$ref` that cannot be found.
+     */
+    protected function resolveAllOfTree(ValidationField $field) {
```

Comment:
We want methods to start out as private rather than protected to enable as much refactoring as possible without breaking BC. We are certainly open to PRs that need to make explicit use of subclassing, but want to be as safe as possible.